### PR TITLE
Part1

### PR DESCRIPTION
**Pull request by hadiys**

### CODE REVIEW

File: main.cpp [hunk 1]
```diff
@@ -76,9 +76,660 @@ Purpose:  This project continues developing Project3.
...
+		ElevatorUser();
+		~ElevatorUser();
+
+		void callElevator();
+        void enterElevator();
+        void setDestination(int newDestinationFloor);
+		void exitElevator();
+		void takeStairsInstead(int newDestinationFloor);
+	};
+};
```
Comment (matkatmusic) [inline main.cpp]: Fix this indentation
- [x] revise

File: main.cpp [hunk 2]
```diff
@@ -76,9 +76,660 @@ Purpose:  This project continues developing Project3.
...
+
+    Building();
+    ~Building();
+
+	void closeBuilding();
+	void openBuilding();
+	bool buildingEmpty();
+	void clearBuilding();
+	void callElevatorAAndTakePassenger(int designatedFloor, double passengerWeight);
+};
```
Comment (matkatmusic) [inline main.cpp]: Fix this indentation
- [x] revise

File: main.cpp [hunk 3]
```diff
@@ -76,9 +76,660 @@ Purpose:  This project continues developing Project3.
...
+			}
+			else
+			{
+				sharpEnough = false;
+				std::cout << "Needs more sharpening" << std::endl;
+			}	
+	}
+	else
+		std::cout << "Pencil sharp enough" << std::endl;
+}
```
Comment (matkatmusic) [inline main.cpp]: Fix this indentation
- [x] revise

**Review by matkatmusic — CHANGES_REQUESTED** [review]
Please review the coding standard, especially the section explaining indentation guidelines.
Your code needs reformatting.
I didn't mark every spot.  All of it needs fixing.
Uncomment everything in main() so that it executes.
- [x] revise

File: main.cpp (hunk 3 above)
Comment (hadiys) [reply main.cpp]: Copying the code from Visual Studio caused this. On replit the code is correctly indented.

File: main.cpp (hunk 2 above)
Comment (hadiys) [reply main.cpp]: Copying the code from Visual Studio caused this. On replit the code is correctly indented.

File: main.cpp [hunk 4]
```diff
@@ -76,9 +76,656 @@ Purpose:  This project continues developing Project3.
...
+{
+    if(isPlugged && drillbitAttached && holes > 0)
+    {
+        std::cout << "Drilling holes..." << std::endl;
+        std::cout << "Drilling done!!" << std::endl;
+    }
+    else if(holes < 1)
+        std::cout << "You can't drill " << holes << " holes..." << std::endl;
+    else 
+        std::cout << "Whoops! Check if your drill is ready" << std::endl;
```
Comment (matkatmusic) [inline main.cpp]: If you use curly braces for the `if()` block, use them for the `else if` and `else` blocks too.
- [x] revise

File: main.cpp [hunk 5]
```diff
@@ -76,9 +76,656 @@ Purpose:  This project continues developing Project3.
...
+        while(numItemsAdded < maxItemsToHold && itemsToAdd > 0)
+        {
+            ++numItemsAdded;
+            --itemsToAdd;
+        }
+        (numItemsAdded == maxItemsToHold) ?  std::cout << "Pencilcase full!" << std::endl : std::cout << numItemsAdded << " items added..." << std::endl;
+    }
+    else
+        std::cout << "Open the pencilcase first" << std::endl;
+}
```
Comment (matkatmusic) [inline main.cpp]: If you use curly braces for the `if()` block, use them for the `else if` and `else` blocks too.
- [x] revise

File: main.cpp [hunk 6]
```diff
@@ -76,9 +76,656 @@ Purpose:  This project continues developing Project3.
...
+        }
+        else
+        {
+            sharpEnough = false;
+            std::cout << "Needs more sharpening" << std::endl;
+        }	
+    }
+    else
+        std::cout << "Pencil sharp enough" << std::endl;
+}
```
Comment (matkatmusic) [inline main.cpp]: If you use curly braces for the `if()` block, use them for the `else if` and `else` blocks too.
- [x] revise

File: main.cpp [hunk 7]
```diff
@@ -76,9 +76,656 @@ Purpose:  This project continues developing Project3.
...
+                itemsToRemove--;
+            }
+            std::cout << numItemsAdded << " items remaining" << std::endl;
+        }
+        else
+            std::cout << "Pencilcase is closed or nothing inside" << std::endl;
+    }
+    else
+        std::cout << "Can't remove " << itemsToRemove << " items"  << std::endl;
+}
```
Comment (matkatmusic) [inline main.cpp]: If you use curly braces for the `if()` block, use them for the `else if` and `else` blocks too.
- [x] revise

File: main.cpp [hunk 8]
```diff
@@ -76,9 +76,656 @@ Purpose:  This project continues developing Project3.
...
+
+void Drill::attachDrillbit(float drillbitWidth)
+{
+    rotationsToSecureBit = 0.8f * drillbitWidth;
+
+    std::cout << "Securing drill bit... " << std::endl;
+
+    for(int i = 0; i < int(rotationsToSecureBit); ++i)
+        chuckTightness+= rotationsToSecureBit;
```
Comment (matkatmusic) [inline main.cpp]: add space around the `+=`
- [x] revise

File: main.cpp [hunk 9]
```diff
@@ -76,9 +76,656 @@ Purpose:  This project continues developing Project3.
...
+    {
+        if(isOpen && numItemsAdded > 0)
+        {
+            std::cout << "Items leaving the pencilcase..." << std::endl;
+
+            while(numItemsAdded > 0 && itemsToRemove > 0)
+            {
+                numItemsAdded--;
+                itemsToRemove--;
+            }
```
Comment (matkatmusic) [inline main.cpp]: use `pre-increment/decrement`
- [x] revise

File: main.cpp [hunk 10]
```diff
@@ -76,9 +76,656 @@ Purpose:  This project continues developing Project3.
...
+    else
+        std::cout << "Pencil sharp enough" << std::endl;
+}
+
+void PencilCase::Pencil::writeSomething(std::string writing)
+{
+    std::cout << writing << std::endl;
+    std::cout << "Finished writing" << std::endl;
+    sharpness-= 0.089f;
+}
```
Comment (matkatmusic) [inline main.cpp]: add space around `-=`
- [x] revise

File: main.cpp [hunk 11]
```diff
@@ -76,9 +76,656 @@ Purpose:  This project continues developing Project3.
...
+void Elevator::ElevatorUser::enterElevator()
+{
+    if(!isInsideElevator)
+    {
+        isInsideElevator = true;
+        std::cout << "You have entered the elevator, now go to a floor" << std::endl;
+    }
+    else
+        std::cout << "You are already inside the elevator" << std::endl;
+}
```
Comment (matkatmusic) [inline main.cpp]: If you use curly braces for the `if()` block, use them for the `else if` and `else` blocks too.
- [x] revise

File: main.cpp [hunk 12]
```diff
@@ -76,9 +76,656 @@ Purpose:  This project continues developing Project3.
...
+    std::cout << "Exiting... ";
+
+    if(isInsideElevator)
+    {
+        isInsideElevator = false;
+        std::cout << "You have exited the elevator" << std::endl;
+    }    
+    else
+        std::cout << "You are not in an elevator yet!" << std::endl;
+}
```
Comment (matkatmusic) [inline main.cpp]: If you use curly braces for the `if()` block, use them for the `else if` and `else` blocks too.
- [x] revise

File: main.cpp [hunk 13]
```diff
@@ -76,9 +76,656 @@ Purpose:  This project continues developing Project3.
...
+
+void Elevator::ElevatorUser::takeStairsInstead(int newDestinationFloor)
+{
+    destinationFloor = newDestinationFloor;
+
+    while(destinationFloor != currentFloor)
+            (destinationFloor > currentFloor) ? ++currentFloor : --currentFloor; 
+
+    std::cout << "You have taken the stairs to floor " << currentFloor << std::endl;
+}
```
Comment (matkatmusic) [inline main.cpp]: Fix this indentation
- [x] revise

File: main.cpp [hunk 14]
```diff
@@ -76,9 +76,656 @@ Purpose:  This project continues developing Project3.
...
+        std::cout << "Packing the pencilcase... Good to go!" << std::endl;
+    }
+    else if(schoolPencilCase.inSchoolBag)
+        std::cout << "Pencilcase already packed" << std::endl;
+    else
+    {
+        schoolPencilCase.removeItems(schoolPencilCase.maxItemsToHold);
+        std::cout << "** You forgot to close your pencilcase!! **" << std::endl;
+    }
+}
```
Comment (matkatmusic) [inline main.cpp]: If you use curly braces for the `if()` block, use them for the `else if` and `else` blocks too.
- [x] revise

File: main.cpp [hunk 15]
```diff
@@ -76,9 +76,656 @@ Purpose:  This project continues developing Project3.
...
+void SchoolBag::unpackPencilCase()
+{
+    if(!schoolPencilCase.inSchoolBag)
+        std::cout << "Pencilcase already unpacked or not in schoolbag" << std::endl;
+    else
+    { 
+        schoolPencilCase.inSchoolBag = false;
+        std::cout << "Unpacked the pencilcase" << std::endl;
+    }
+}
```
Comment (matkatmusic) [inline main.cpp]: If you use curly braces for the `else()` block, use them for the `if` block too.
- [x] revise

File: main.cpp [hunk 16]
```diff
@@ -76,9 +76,656 @@ Purpose:  This project continues developing Project3.
...
+    if(buildingEmpty() && isOpen)
+    {
+        isOpen = false;
+        std::cout << "Closing building. Elevators clear..." << std::endl;
+    }
+    else if(!isOpen)
+        std::cout << "Building already closed" << std::endl;
+    else
+        std::cout << "Cannot close building now.. There are still users in the elevator" << std::endl;
+}
```
Comment (matkatmusic) [inline main.cpp]: If you use curly braces for the `if()` block, use them for the `else if` and `else` blocks too.
- [x] revise

File: main.cpp [hunk 17]
```diff
@@ -76,9 +76,656 @@ Purpose:  This project continues developing Project3.
...
+void Building::openBuilding()
+{
+    if(!isOpen)
+    {
+        isOpen = true;
+        std::cout << "Building is now open" << std::endl;
+    }
+    else
+        std::cout << "Building is already open" << std::endl;
+}
```
Comment (matkatmusic) [inline main.cpp]: If you use curly braces for the `if()` block, use them for the `else if` and `else` blocks too.
- [x] revise

File: main.cpp [hunk 18]
```diff
@@ -76,9 +76,656 @@ Purpose:  This project continues developing Project3.
...
+{
+    if(isOpen)
+    {
+        std::cout << "Calling elevator " << elevatorA.name << std::endl;
+        elevatorA.goToFloor(designatedFloor);
+        elevatorA.onboardPassenger(userWeight);
+    }
+    else	
+        std::cout << "Cannot call elevator, building not open yet" << elevatorA.name << std::endl;
+}
```
Comment (matkatmusic) [inline main.cpp]: If you use curly braces for the `if()` block, use them for the `else if` and `else` blocks too.
- [x] revise

File: main.cpp [hunk 19]
```diff
@@ -76,9 +76,656 @@ Purpose:  This project continues developing Project3.
...
+	b1.clearBuilding();
+
+	Building b2;
+
+	b2.openBuilding();
+	b2.callElevatorAAndTakePassenger(51, 100);
+	b2.clearBuilding();
+	b2.callElevatorAAndTakePassenger(44, 120);
+	b2.openBuilding();
+	b2.clearBuilding();
```
Comment (matkatmusic) [inline main.cpp]: Fix this indentation
remove these unnecessary blank lines
- [x] revise

**Review by matkatmusic — CHANGES_REQUESTED** [review]
Your code still has some formatting issues as well as some style inconsistencies.
It's important to catch these issues now before they become permanent habits.
The Readme.md states that you should look at the Files Tab of the Pull Request, so you can see the same thing I'll see before I get a chance to look at it.   The purpose of this step is so you can catch these issues before I do and for you to correct them before I request the changes.
- [x] revise

**Review by matkatmusic — APPROVED** [review]

**Pull request merged**